src/test/java/store/service/SaleServiceTest.java

```diff
@@ -1,227 +1,282 @@
+        when(saleCriteriaQuery.from(Sale.class)).thenReturn(saleRoot);
+        when(saleRoot.get(Sale_.id)).thenReturn(path);
+        when(entityManager.createQuery(saleCriteriaQuery)).thenReturn(typedQuery);
+        when(typedQuery.getResultList()).thenReturn(list);
+
+        //then
+        List<SaleDTO> resultListDTO = saleService.get(idParameter, saleDateParameter, categoriesParameter);
+        List<Sale> resultList = resultListDTO.stream().map(saleDTO1 -> mapper.toEntity(saleDTO1, Sale.class)).collect(Collectors.toList());
+
+        //testing the method is calling
+        verify(typedQuery).getResultList();
+
+        //testing correct data
+        assertEquals(list, resultList);
+        assertEquals(2, resultListDTO.size());
+    }
+
+    @Test
+    public void testGetIfPassIdParameter(){
+        when(entityManager.getCriteriaBuilder()).thenReturn(criteriaBuilder);
+        when(criteriaBuilder.createQuery(Sale.class)).thenReturn(saleCriteriaQuery);
+        when(saleCriteriaQuery.from(Sale.class)).thenReturn(saleRoot);
+        when(saleRoot.get(Sale_.id)).thenReturn(path);
+        when(entityManager.createQuery(saleCriteriaQuery)).thenReturn(typedQuery);
+        when(typedQuery.getResultList()).thenReturn(saleList);
+
+        List<SaleDTO> resultListDTO = saleService.get(ID, saleDateParameter, categoriesParameter);
+        //convert saleDTO into sale
+        List<Sale> resultList = resultListDTO.stream().map(saleDTO -> mapper.toEntity(saleDTO, Sale.class)).collect(Collectors.toList());
+
+        //testing the method is calling
+        verify(typedQuery).getResultList();
+
+        //testing correct data
+        assertEquals(saleList, resultList);
+        assertEquals(1, resultList.size());
+    }
+
+    @Test
+    public void testIfPassSaleDateParameter(){
+        //given
+        LocalDate date = LocalDate.of(YEAR, 4, DAY);
+
+        //when
+        when(entityManager.getCriteriaBuilder()).thenReturn(criteriaBuilder);
+        when(criteriaBuilder.createQuery(Sale.class)).thenReturn(saleCriteriaQuery);
+        when(saleCriteriaQuery.from(Sale.class)).thenReturn(saleRoot);
+        when(saleRoot.get(Sale_.id)).thenReturn(path);
+        when(entityManager.createQuery(saleCriteriaQuery)).thenReturn(typedQuery);
+        when(typedQuery.getResultList()).thenReturn(saleList);
+
+        //then
+        List<SaleDTO> resultListDTO = saleService.get(idParameter, date, categoriesParameter);
+        List<Sale> resultList = resultListDTO.stream().map(saleDTO -> mapper.toEntity(saleDTO, Sale.class)).collect(Collectors.toList());
+
+        //testing the method is calling
+        verify(typedQuery).getResultList();
+
+        //checking correct data
+        assertEquals(saleList, resultList);
+        assertEquals(1, resultList.size());
+    }
+
+    @Test
+    public void testIfPassCategoriesParameter(){
+        //given
+        List<String> cat = Arrays.asList("First", "Second");
```

Comment:
why you need to create an additional list if you already have such global variable "categories"